src/testing/react/__tests__/MockedProvider.test.tsx

```diff
@@ -481,6 +481,105 @@ describe("General use", () => {
     expect(errorThrown).toBeFalsy();
   });
 
+  it('should reuse mocks when instructed to', async () => {
+    let isLoaded = false;
+    let isError = false;
+    function Component({ username }: Variables) {
+      const { loading, error } = useQuery<Data, Variables>(query, { variables: { username }, fetchPolicy: 'network-only' });
+      if (!loading) {
+        if (error) {
+          isError = true;
+        }
+        isLoaded = true;
+      }
+      return null;
+    }
```

Comment:
```suggestion
    interface Result {
      current: QueryResult | null
    }
    
    const result: Result = { current: null };
    
    function Component({ username }: Variables) {
      result.current = useQuery<Data, Variables>(query, { variables: { username }, fetchPolicy: 'network-only' });

      return null;
    }
```

Rather than using primitive variables here, a trick I like to use is to create an object with a `current` property that can be assigned inside the component. This allows you to remove the conditional logic you have entirely and instead can check against the raw values directly. As mentioned in a previous comment, this also makes it so that you can check against a specific error that was returned, rather than the possibility that the error could be anything.

With this change, your `waitForLoaded` would look like this following:

```ts
const waitForLoaded = async () => {
  await waitFor(() => {
    expect(result.current.loading).toBe(false);
    expect(result.current.error).toBeUndefined();
  });
}
```